src/pu-package.c

```diff
@@ -0,0 +1,253 @@
+    guint64 child_size;
+    const gchar *file_attr = G_FILE_ATTRIBUTE_STANDARD_NAME ","
+                             G_FILE_ATTRIBUTE_STANDARD_SIZE;
+
+    dir_path = g_file_get_path(dir);
+    pu_str_pre_remove(dir_path, prefix_len);
+    if (strlen(dir_path))
+        g_print("%s/\n", dir_path);
+    dir_enum = g_file_enumerate_children(dir, file_attr, G_FILE_QUERY_INFO_NONE, NULL, error);
+    if (!dir_enum) {
+        g_set_error(error, PU_PACKAGE_ERROR, PU_PACKAGE_ERROR_ITER_FAILED,
+                    "Failed creating enumerator for '%s'",
+                    g_file_get_path(dir));
+        return FALSE;
+    }
+
+    while (TRUE) {
+        GFileInfo *child_info;
+        g_autofree gchar *child_path = NULL;
+
+        if (!g_file_enumerator_iterate(dir_enum, &child_info, NULL, NULL, error)) {
+            g_set_error(error, PU_PACKAGE_ERROR, PU_PACKAGE_ERROR_ITER_FAILED,
+                        "Failed iterating file enumerator for '%s'",
+                        g_file_get_path(dir));
+            return FALSE;
+        }
+
+        if (!child_info)
+            break;
+
+        child_path = g_build_filename(g_file_get_path(dir), g_file_info_get_name(child_info), NULL);
+        if (g_file_test(child_path, G_FILE_TEST_IS_DIR) && recursive) {
+            g_autoptr(GFile) subdir = g_file_new_for_path(child_path);
+
+            if (!print_dir_content(subdir, TRUE, print_size, error))
+                return FALSE;
+        } else {
+            pu_str_pre_remove(child_path, prefix_len);
+            g_print("%s", child_path);
+            if (print_size) {
+                child_size = g_file_info_get_size(child_info);
+                g_print(" (%s)", g_format_size(child_size));
+            }
+            g_print("\n");
+        }
+    }
+
+    return TRUE;
+}
+
+gboolean
+pu_package_list_contents(const gchar *package,
+                         gboolean print_size,
+                         GError **error)
+{
+    g_autofree gchar *dir_basename = NULL;
+    g_autofree gchar *mountpoint = NULL;
+    g_autofree gchar *layout_file = NULL;
+    g_autoptr(GFile) dir = NULL;
+
+    g_return_val_if_fail(g_strcmp0(package, "") > 0, FALSE);
+    g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+    if (!g_file_test(package, G_FILE_TEST_IS_REGULAR)) {
+        g_set_error(error, PU_PACKAGE_ERROR, PU_PACKAGE_ERROR_NOT_FOUND,
+                    "'%s' is not a regular file", package);
+        return FALSE;
+    }
+
+    dir_basename = g_strdup_printf("%s-%08x", PU_PACKAGE_BASENAME, g_random_int());
+    mountpoint = pu_create_mount_point(dir_basename, error);
+    if (!mountpoint)
+        return FALSE;
+
+    if (!pu_mount(package, mountpoint, "squashfs", "loop,ro", error))
+        return FALSE;
+
+    layout_file = pu_package_get_layout_file(mountpoint, error);
+    if (!layout_file) {
+        g_prefix_error(error, "Invalid package: ");
+        pu_umount(mountpoint, NULL);
+        return FALSE;
+    }
```

Comment:
Use `pu_package_mount()` here as it does exactly the same.

Reply:
Done. Also used `pu_package_umount()` to simplify this function.